newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -137,219 +69,192 @@ type State = {|
+            onInputChange={handleInputChange}
+            options={this.props.dataSource}
+            renderOption={renderItem}
+            getOptionDisabled={this.isOptionDisabled}
+            ListboxProps={{
+              style: {
+                padding: 0,
+                margin: 0,
+              },
             }}
-            itemToString={item =>
-              item ? (item.type === 'separator' ? '' : item.value) : ''
-            }
-          >
-            {({
-              getInputProps,
-              getItemProps,
-              getLabelProps,
-              getMenuProps,
-
-              // Actions:
-              closeMenu,
-              openMenu,
-
-              // State:
-              highlightedIndex,
-              inputValue,
-              isOpen,
-              selectedItem,
-            }) => {
-              const { onBlur, ...inputProps } = getInputProps({
+            getOptionLabel={getOptionLabel}
+            filterOptions={filterFunction}
+            renderInput={params => {
+              var { InputProps, ...other } = getDefaultStylingProps(params);
+              InputProps = {
+                ...InputProps,
                 placeholder:
-                  typeof props.hintText === 'string'
-                    ? props.hintText
-                    : i18n._(props.hintText),
-                disabled: props.disabled,
-              });
-
-              // Wrap onBlur to close the menu and commit the changes to the value
-              const wrappedOnBlur = event => {
-                onBlur(event);
-
-                // Downshift will, after the blur event, reset the state so that the
-                // input value is itemToString applied to the selected item - which can
-                // be null as we allow whatever value is entered, even an incomplete one.
-                // Use a setTimeout to clear the inputValue just after Downshift has
-                // changed inputValue.
-                // (this is purely "visual", the onChange props is properly called, this
-                // being done or not)
-                setTimeout(() => {
-                  this.setState({
-                    inputValue: null,
-                  });
-                });
-
-                // Also close the menu
-                closeMenu();
-
-                // Call onChange with whatever value is entered, even an incomplete one.
-                props.onChange(event.currentTarget.value);
-
-                if (props.onBlur) props.onBlur(event);
+                  typeof this.props.hintText === 'string'
+                    ? this.props.hintText
+                    : i18n._(this.props.hintText),
               };
-
-              const onFocus = event => {
-                if (props.openOnFocus) openMenu();
-                this.setState({
-                  inputValue: props.value,
-                });
-              };
-
+              const defaultStylingProps = { InputProps, ...other };
```

Comment:
Not sure if creating this object is worth it, just put the content in the `<TextField` props directly.